### PR TITLE
feat: check for cySync and device update on regular interval

### PR DESCRIPTION
**Pull request by jhasuraj01**

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6292 Cysync is not notifying regarding updates for both cysync and device in opened state](https://app.clickup.com/t/9002019994/PRF-6292)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6292 Cysync is not notifying regarding updates for both cysync and device in opened state](https://app.clickup.com/t/9002019994/PRF-6292)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6292 Cysync is not notifying regarding updates for both cysync and device in opened state](https://app.clickup.com/t/9002019994/PRF-6292)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6292 Cysync is not notifying regarding updates for both cysync and device in opened state](https://app.clickup.com/t/9002019994/PRF-6292)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6292 Cysync is not notifying regarding updates for both cysync and device in opened state](https://app.clickup.com/t/9002019994/PRF-6292)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6292 Cysync is not notifying regarding updates for both cysync and device in opened state](https://app.clickup.com/t/9002019994/PRF-6292)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6292 Cysync is not notifying regarding updates for both cysync and device in opened state](https://app.clickup.com/t/9002019994/PRF-6292)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6292 Cysync is not notifying regarding updates for both cysync and device in opened state](https://app.clickup.com/t/9002019994/PRF-6292)

File: packages/cysync-core/src/context/appUpdate.tsx [hunk 1]
```diff
@@ -71,6 +71,15 @@ export const AppUpdateProvider: React.FC<AppUpdateProviderProps> = ({
...
   );
 
+  useEffect(() => {
+    const minutes = 15;
+    const checkUpdateInterval = setInterval(
+      checkForUpdates,
+      minutes * 60 * 1000,
+    );
+    return () => clearInterval(checkUpdateInterval);
+  }, [checkForUpdates]);
```
Comment (irshadCypherock) [inline packages/cysync-core/src/context/appUpdate.tsx:81]: If useEffect depends on a function, we need to make sure that function does not change on re-render. Use `useCallback` for that.

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/context/appUpdate.tsx (hunk 1 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/context/appUpdate.tsx:81]: Done

File: packages/cysync-core/src/hooks/useAppUpdate.ts [hunk 2]
```diff
@@ -107,7 +107,13 @@ export const useAppUpdate = () => {
     } finally {
       setIsUpdatesChecked(true);
     }
-  };
+  }, [
+    setIsUpdatesChecked,
+    setAppUpdateState,
+    autoUpdater.checkForUpdates,
+    setUpdateInfo,
+    onError,
```
Comment (irshadCypherock) [inline packages/cysync-core/src/hooks/useAppUpdate.ts]: No need for dependencies here, since there are not state variables

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/hooks/useAppUpdate.ts (hunk 2 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/hooks/useAppUpdate.ts]: removed dependencies.

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**